internal/pkg/util/fs/overlay/overlay_item_linux.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2019-2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package overlay
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/sylabs/singularity/internal/pkg/util/bin"
+	"github.com/sylabs/singularity/pkg/image"
+	"github.com/sylabs/singularity/pkg/sylog"
+)
+
+// Item represents information about a single overlay item (as specified,
+// for example, in a single --overlay argument)
+type Item struct {
+	// Type represents what type of overlay this is (from among the values in pkg/image)
+	Type int
+
+	// Writable represents whether this is a writable overlay
+	Writable bool
+
+	// BarePath is the path of the overlay stripped of any colon-prefixed
+	// options (like ":ro")
+	BarePath string
```

Comment:
`SourcePath` would probably be clearer when it is subsequently used in code. We're starting to get quite overloaded r.e. the concept of `path of the overlay`, and `Bare` isn't very informative.

In comments here, prefer talking about the `overlay Item`, and not just `overlay`.